¿Qué números de Fibonacci menores que 10²⁰ son cuadrados?


In [3]:
from sympy import sqrt

def es_potencia(n, pot = 2):
    return int(pow(n,1/pot)) ** pot == n

def potencias_fibonacci(n, pot = 2):
    fibs = []
    a, b = 0, 1
    while a < n:
        if es_potencia(a,pot):
            fibs.append(a)
        a, b = b, a+b
    return fibs

potencias_fibonacci(10**20)

[0, 1, 1, 144]


¿Qué números de Fibonacci menores que 10²⁰ son cubos?

In [4]:
potencias_fibonacci(10**20, pot = 3)

[0, 1, 1, 8]

¿Qué números de Fibonacci menores que 10²⁰ son cuartas potencias?

In [5]:
potencias_fibonacci(10**20, pot = 4)

[0, 1, 1]

El artículo relacionado a la prueba de que 0, 1, 8 y 144 son los únicos números de Fibonacci que son potencias perfectas se puede encontrar en el siguiente link: https://annals.math.princeton.edu/wp-content/uploads/annals-v163-n3-p05.pdf.